src/app/modules/main/profile_section/keycard/module.nim

```diff
@@ -157,52 +161,68 @@ method runCreatePukPopup*(self: Module) =
 method runCreateNewPairingCodePopup*(self: Module) =
   info "TODO: Create New Pairing Code for a Keycard..."
 
-proc buildKeycardList(self: Module) =
+proc buildKeycardItem(self: Module, walletAccounts: seq[WalletAccountDto], keyPair: KeyPairDto): KeycardItem =
   let findAccountByAccountAddress = proc(accounts: seq[WalletAccountDto], address: string): WalletAccountDto =
     for i in 0 ..< accounts.len:
       if(accounts[i].address == address):
         return accounts[i]
     return nil
 
-  let accounts = self.controller.getWalletAccounts()
+  var knownAccounts: seq[WalletAccountDto]
+  for accAddr in keyPair.accountsAddresses:
+    let account = findAccountByAccountAddress(walletAccounts, accAddr)
+    if account.isNil:
+      ## we should never be here cause we need to remove deleted accounts from the `keypairs` table and sync
```

Comment:
How about logging a warning to help debug if you think there is a chance to reach this error case?

Reply:
@stefandunca at the moment it's not reachable cause we don't have a keycard part added to the syncing, but once we do it, this may be reached if user deletes a wallet account on one instance but appropriate account from the keypairs table is not deleted or if all that is done, but not synced properly.